### PR TITLE
[0.18] Backport: Adding global TLS/SASL for the KafkaChannel

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

bringing sandbox PR to here:
https://github.com/knative-sandbox/eventing-kafka/pull/211

### CODE REVIEW

File: kafka/channel/pkg/utils/util.go [hunk 1]
```diff
@@ -29,6 +34,8 @@ import (
 
 const (
 	BrokerConfigMapKey           = "bootstrapServers"
+	TlsName                      = "tlsName"
+	TlsNamespace                 = "tlsNamespace"
```
Comment (matzew) [inline kafka/channel/pkg/utils/util.go:38]: Oh, I think we should name these new `ConfigMap` keys:

* `AuthSecretName`
* `AuthSecretNamespace`

**Comment by matzew** [comment]
/assign @lberk 
/assign @aliok 

Can one of you take a looksie?

File: kafka/channel/pkg/utils/util.go (hunk 1 above)
Comment (lionelvillard) [reply kafka/channel/pkg/utils/util.go:38]: Or `TlsSecretName`.

**Comment by lionelvillard** [comment]
Why are you opening a PR here vs backporting a yet-to-be-done PR in knative-sandbox?

File: kafka/channel/pkg/utils/util.go (hunk 1 above)
Comment (matzew) [reply kafka/channel/pkg/utils/util.go:38]: `TlsSecretName` -> it supports both: TLS and SASL

**Comment by matzew** [comment]
> Why are you opening a PR here vs backporting a yet-to-be-done PR in knative-sandbox?

Just a matter of where I started :-)  I will port forward, on Monday - or later :-)

File: kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go [hunk 2]
```diff
@@ -98,6 +98,8 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		KnCEConnectionArgs: connectionArgs,
 		ClientID:           "kafka-ch-dispatcher",
 		Brokers:            kafkaConfig.Brokers,
+		TlsName:            kafkaConfig.TlsName,
+		TlsNamespace:       kafkaConfig.TlsNamespace,
```
Comment (aliok) [inline kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go:102]: I think as others suggested, we should change the name.
`TlsSecretName/TlsSecretNamespace`?

File: kafka/channel/pkg/utils/util.go (hunk 1 above)
Comment (aliok) [reply kafka/channel/pkg/utils/util.go:38]: I was thinking `TlsSecretName` but `AuthSecretName` sounds better actually

File: kafka/client.go [hunk 3]
```diff
@@ -125,6 +127,31 @@ func newTLSConfig(clientCert, clientKey, caCert string) (*tls.Config, error) {
 	return config, nil
 }
 
+func MakeAdminClient(clientID string, kafkaAuthCfg *utils.KafkaAuthConfig, bootstrapServers []string) (sarama.ClusterAdmin, error) {
```
Comment (aliok) [inline kafka/client.go:130]: Content of this looks very similar to the block at https://github.com/knative/eventing-contrib/pull/1667/files#diff-27083a6c5d0b81eb155bf1deda5b4231e49fe42631729db2f8c7b3b55eb57d0fR92-R114

If possible, reuse the logic?

File: kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go (hunk 2 above)
Comment (aliok) [reply kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go:102]: Oh, just saw `AuthSecretName` suggestion. That sounds better

**Comment by aliok** [comment]
/lgtm
/approve

Ignoring linting check

**Comment by matzew** [comment]
> Ignoring linting check

I have no idea what's wrong here... tbh

**Pull request merged**